include/fmt/format.h

```diff
@@ -779,6 +783,40 @@ class truncating_iterator {
   reference operator*() const { return count_ < limit_ ? *out_ : blackhole_; }
 };
 
+template <typename OutputIt>
+class truncating_iterator<OutputIt, std::true_type> {
+ private:
+  OutputIt out_;
+  std::size_t limit_;
+  std::size_t count_;
+
+ public:
+  typedef std::output_iterator_tag iterator_category;
+  typedef typename OutputIt::container_type::value_type value_type;
+  typedef void difference_type;
+  typedef void pointer;
+  typedef void reference;
+  typedef truncating_iterator _Unchecked_type;  // Mark iterator as checked.
+
+  truncating_iterator(OutputIt out, std::size_t limit)
+    : out_(out), limit_(limit), count_(0) {}
+
+  OutputIt base() const { return out_; }
+  std::size_t count() const { return count_; }
+
+  truncating_iterator& operator=(value_type val) {
+    if (count_++ < limit_)
+      out_ = val;
+    return *this;
+  }
+
+  truncating_iterator& operator++() { return *this; }
+
+  truncating_iterator& operator++(int) { return *this; }
+
+  truncating_iterator& operator*() { return *this; }
+};
```

Comment:
Is it possible to move common code between the two specializations into a superclass to reduce copy-paste?